### ISSUE TITLE
Simple Rasa Core + NLU Rest API delivering responses

### ISSUE THREAD

**Issue by jenniferzhu** [RasaHQ/rasa_nlu#1863]
My goal is to wrap the rasa core + nlu bot into an API so that I can call it in some JavaScript-based UI, such as a simple angularjs-chat ( http://angularjs.chat/tutorials/angularjs-basic-chat-module/).

 However, I can only use CustomInput to send a message to "localhost:5005" by POST but I cannot get the response from my bot. Is there any way that I could build/modify on top of the rasa_core? I know that it is already using Flask, so I don’t want to build another Flask app on top of that.

In summary, I guess it would be great if we can set up an API such as "localhost:5005" to receive the questions from users and send responses back to the same channel, like what we have using the consoleInput.

**Comment by pradeepd80** [comment]
Hi All ,
Could you please help me any one I am facing issue on chatbot conversation . My UI in angular and Now i want to access rasa core + nlu as rest from my angular services.

my python complete code as follows: 
I am new in python and rasa: 

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

import logging

import requests

#from rasa_core.utils import EndpointConfig
from rasa_core.channels import HttpInputChannel
from rasa_core import utils
from rasa_core.agent import Agent
from rasa_core.interpreter import RasaNLUInterpreter
from rasa_core.channels.channel import UserMessage
from rasa_core.channels.direct import CollectingOutputChannel
from rasa_core.channels.rest import HttpInputComponent
from flask import Blueprint, request, Flask    # jsonify,
from flask_cors import CORS, cross_origin
from flask_restful import Resource, Api
from json import dumps
from flask_jsonpify import jsonify



app = Flask(__name__)
api = Api(app)

CORS(app)

logger = logging.getLogger(__name__)

class SimpleWebBot(HttpInputChannel):
	"""A simple web bot that listens on a url and responds. """	
	def blueprint(self, on_new_message):		
		custom_webhook = Blueprint('custom_webhook', __name__)
		
		@custom_webhook.route("/status", methods=['GET'])
		def health():	
			print(f"1***********{request}**********************")
			return jsonify({"status":"ok pradeep find out status by localhost:5004/chat/status Service"})
					
		@custom_webhook.route("/", methods=['POST'])		
		def receive():
			print(f"2***********{request}**********************")
			payload = request.json
			print(f"3***********************{payload}************************")
			sender_id = payload.get("sender", None)
			text = payload.get("message", None)
			out = CollectingOutputChannel()
			print("11***********************************{out}*********************************")
			on_new_message(UserMessage(text, out, sender_id))
			responses = [m for _, m in out.messages]			
			return jsonify(responses)
		
		return custom_webhook
		
def run(serve_forever=True):
	interpreter = RasaNLUInterpreter("./models/nlu/default/weathernlu")	
	agent = Agent.load("./models/dialogue", interpreter=interpreter)	
	input_channel = SimpleWebBot(5004, "/chat") 	
	if serve_forever:
		print("6**run*********************SERVER STARTED*********************************************")		
		agent.handle_channel(HttpInputChannel(5004, "/chat",input_channel))		
	return agent
	
if __name__ == '__main__':	
	utils.configure_colored_logging(loglevel="INFO")	
	run()
		
Please be response I stucked. 
Thanks Adv.

**Comment by pradeepd80** [comment]
I am facing mainly two issue
1. In postman is ok. but  when I am calling "http://127.0.0.1:5004/chat/status" from angular then showing error:
Access to XMLHttpRequest at 'http://127.0.0.1:5004/chat/status' from origin 'http://localhost:4200' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

2. When i am calling http://127.0.0.1:5004/chat from postman there is showing server error.

Please help.
Thanks in Adv.
